lib/realtime/block_waiver.ex

```diff
@@ -98,15 +98,35 @@ defmodule Realtime.BlockWaiver do
   @spec within_60_minutes?(Util.Time.time_of_day(), Util.Time.time_of_day()) :: boolean
   defp within_60_minutes?(time, previous_time), do: time - previous_time <= 3600
 
-  @spec from_trip_stop_time_waivers([trip_stop_time_waiver()]) :: t()
-  def from_trip_stop_time_waivers(trip_stop_time_waivers) do
+  @spec from_trip_stop_time_waivers([trip_stop_time_waiver()], Date.t()) :: t()
+  def from_trip_stop_time_waivers(trip_stop_time_waivers, date_for_block) do
     {start_time, %StopTimeUpdate{remark: remark}} = List.first(trip_stop_time_waivers)
     {end_time, _} = List.last(trip_stop_time_waivers)
 
     %__MODULE__{
-      start_time: start_time,
-      end_time: end_time,
+      start_time: Util.Time.timestamp_for_time_of_day(start_time, date_for_block),
+      end_time: Util.Time.timestamp_for_time_of_day(end_time, date_for_block),
       remark: remark
     }
   end
+
+  @spec date_for_block(Block.t()) :: Date.t()
+  defp date_for_block(block) do
+    active_blocks_fn = Application.get_env(:realtime, :active_blocks_fn, &Gtfs.active_blocks/2)
+
+    now = Util.Time.now()
+    ten_minutes = 10 * 60
+    ten_minutes_ago = now - ten_minutes
+    in_ten_minutes = now + ten_minutes
+
+    {date, _blocks} =
+      ten_minutes_ago
+      |> active_blocks_fn.(in_ten_minutes)
+      |> Enum.find(
+        {Util.Time.today(), []},
```

Comment:
If we wanted to be stricter about mocking times, we could do `now = Application.get_env(:skate, :now_fn).()` above and then here could be `Util.Time.date_of_timestamp(now)`.

It looks like in this case the tests will still work since they also use `Util.Time.today()` in a matching way, so this is more a general feeling of not wanting to depend on using real times in tests than this causing a specific problem.

Reply:
I'm not worried about that.